src/App.jsx

```diff
@@ -14,45 +19,80 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortCondition, reverseCondition }) {
+  const preparedGoods = [...goods];
+
+  if (sortCondition) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortCondition) {
+        case 'alphabet':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseCondition) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortCondition, setSortCondition] = useState('');
+  const [reverseCondition, setReverseCondition] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer, { sortCondition, reverseCondition },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortCondition(SORT_ALPHABET)}
+          className={`button is-info ${sortCondition === SORT_ALPHABET ? '' : 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortCondition(SORT_LENGTH)}
+          className={`button is-success ${sortCondition === SORT_LENGTH ? '' : 'is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setReverseCondition(!reverseCondition)}
+          className={`button is-warning ${reverseCondition === true ? '' : 'is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {(sortCondition === SORT_LENGTH || sortCondition === SORT_ALPHABET
+        || reverseCondition === true) && (
```

Comment:
```suggestion
        {(sortCondition || reverseCondition) && (
```

Let's simplify your condition a bit